localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/method_request.py

```diff
@@ -113,14 +113,16 @@ def _get_missing_required_parameters(method: Method, request: InvocationRequest)
         if not (request_parameters := method.get("requestParameters")):
             return missing_params
 
+        case_sensitive_headers = list(request.get("headers").keys())
```

Comment:
I know they say `set()` is meant to be faster, but every time I tried to `timeit`, it turned out to be slightly slower. Maybe because I am working with smaller sets? It might have a bigger impact on greater sets? :thinking:

Reply:
Just gave it a try with a list of parameters of 7, and 8 headers, and it's about 15% faster. I guess it can depend heavily 😄